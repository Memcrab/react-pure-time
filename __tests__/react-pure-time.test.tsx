/**
 * @jest-environment jsdom
 */

import * as React from "react";
import * as renderer from "react-test-renderer";
import Time from "../src/react-pure-time";

import {
  mockedStringHour,
  mockedStringMinutes,
} from "./helpers/data-mocked.const";

jest.setTimeout(100000);
describe("test async behavior of the Time component", () => {
  it("Render Time component with relative time", (done) => {
    const TimeComponent = renderer.create(
      <Time value={mockedStringMinutes} relativeTime />
    );
    renderer.act(() => {
      setTimeout(() => {
        expect(TimeComponent.toJSON()).toMatchSnapshot();
        done();
      }, 1000);
    });
  });

  it("1 minute change to hour", (done) => {
    const TimeComponent = renderer.create(
      <Time value={mockedStringHour} relativeTime />
    );
    renderer.act(() => {
      setTimeout(() => {
        expect(TimeComponent.toJSON()).toMatchSnapshot();
        done();
      }, 61000);
    });
  });
});

describe("Time component behavior", () => {
  it("Renders default placeholder with empty props", () => {
    const TimeComponent = renderer.create(<Time />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders default placeholder with UTC flag", () => {
    const TimeComponent = renderer.create(<Time utc />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders custom placeholder with empty value", () => {
    const TimeComponent = renderer.create(
      <Time placeholder="no time defined" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders custom placeholder with UTC flag", () => {
    const TimeComponent = renderer.create(
      <Time placeholder="no time defined" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders date value with default format", () => {
    const TimeComponent = renderer.create(<Time value={1261322332312} />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders date value with custom format H:i:s", () => {
    const TimeComponent = renderer.create(
      <Time value={1261322332312} format="H:i:s" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders date value with custom format H:i:s and UTC flag", () => {
    const TimeComponent = renderer.create(
      <Time value={1261322332312} format="H:i:s" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders right with milliseconds value", () => {
    const TimeComponent = renderer.create(
      <Time value={1261322332312} format="d.m.Y H:i:s" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders right with milliseconds value and UTC", () => {
    const TimeComponent = renderer.create(
      <Time value={1261322332312} format="d.m.Y H:i:s" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders right with object value", () => {
    const TimeComponent = renderer.create(
      <Time value={new Date(1261322332312)} format="d.m.Y H:i:s" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders right with object value and UTC", () => {
    const TimeComponent = renderer.create(
      <Time value={new Date(1261322332312)} format="d.m.Y H:i:s" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Renders right with string 1990-06-16 13:22:17", () => {
    const TimeComponent = renderer.create(
      <Time value="1990-06-16 13:22:17" format="d.m.Y H:i:s" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it('Renders right with string "1990-06-16 13:22:17" and UTC', () => {
    const TimeComponent = renderer.create(
      <Time value="1990-06-16 13:22:17" format="d.m.Y H:i:s" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it('Renders right with string "September 2, 2003, 2:26 am"', () => {
    const TimeComponent = renderer.create(
      <Time value="September 2, 2003, 2:26 am" format="d.m.Y H:i:s" />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it('Renders right with string "September 2, 2003, 2:26 am" and UTC', () => {
    const TimeComponent = renderer.create(
      <Time value="September 2, 2003, 2:26 am" format="d.m.Y H:i:s" utc />
    );
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Checks value and renders placeholder if its invalid", () => {
    const TimeComponent = renderer.create(<Time value="2003 36 2003" />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Checks value and renders placeholder if its invalid with UTC", () => {
    const TimeComponent = renderer.create(<Time value="2003 36 2003" utc />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });

  it("Checks value and renders custom placeholder if its invalid", () => {});

  it("Applies custom styles", () => {
    const TimeComponent = renderer.create(<Time className="customName" />);
    expect(TimeComponent.toJSON()).toMatchSnapshot();
  });
});
