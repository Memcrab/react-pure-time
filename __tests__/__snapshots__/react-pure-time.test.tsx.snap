// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time component behavior Applies custom styles 1`] = `
<span
  className="customName"
>
  —
</span>
`;

exports[`Time component behavior Checks value and renders placeholder if its invalid 1`] = `
<span
  className=""
>
  —
</span>
`;

exports[`Time component behavior Checks value and renders placeholder if its invalid with UTC 1`] = `
<span
  className=""
>
  —
</span>
`;

exports[`Time component behavior Renders custom placeholder with UTC flag 1`] = `
<span
  className=""
>
  no time defined
</span>
`;

exports[`Time component behavior Renders custom placeholder with empty value 1`] = `
<span
  className=""
>
  no time defined
</span>
`;

exports[`Time component behavior Renders date value with custom format H:i:s 1`] = `
<span
  className=""
>
  17:18:52
</span>
`;

exports[`Time component behavior Renders date value with custom format H:i:s and UTC flag 1`] = `
<span
  className=""
>
  15:18:52
</span>
`;

exports[`Time component behavior Renders date value with default format 1`] = `
<span
  className=""
>
  20.12.2009 17:18
</span>
`;

exports[`Time component behavior Renders default placeholder with UTC flag 1`] = `
<span
  className=""
>
  —
</span>
`;

exports[`Time component behavior Renders default placeholder with empty props 1`] = `
<span
  className=""
>
  —
</span>
`;

exports[`Time component behavior Renders right with milliseconds value 1`] = `
<span
  className=""
>
  20.12.2009 17:18:52
</span>
`;

exports[`Time component behavior Renders right with milliseconds value and UTC 1`] = `
<span
  className=""
>
  20.12.2009 15:18:52
</span>
`;

exports[`Time component behavior Renders right with object value 1`] = `
<span
  className=""
>
  20.12.2009 17:18:52
</span>
`;

exports[`Time component behavior Renders right with object value and UTC 1`] = `
<span
  className=""
>
  20.12.2009 15:18:52
</span>
`;

exports[`Time component behavior Renders right with string "1990-06-16 13:22:17" and UTC 1`] = `
<span
  className=""
>
  16.06.1990 09:22:17
</span>
`;

exports[`Time component behavior Renders right with string "September 2, 2003, 2:26 am" 1`] = `
<span
  className=""
>
  02.09.2003 02:26:00
</span>
`;

exports[`Time component behavior Renders right with string "September 2, 2003, 2:26 am" and UTC 1`] = `
<span
  className=""
>
  01.09.2003 23:26:00
</span>
`;

exports[`Time component behavior Renders right with string 1990-06-16 13:22:17 1`] = `
<span
  className=""
>
  16.06.1990 13:22:17
</span>
`;

exports[`test async behavior of the Time component 1 minute change to hour 1`] = `
<span
  className=""
>
   1 hour ago
</span>
`;

exports[`test async behavior of the Time component Render Time component with relative time 1`] = `
<span
  className=""
>
   45 minutes ago
</span>
`;
